### PR TITLE
Create OWNERS files for remaining subdirs of kubernetes/

### LINKED ISSUE #44420: Create OWNERS files for all subdirs of kubernetes/
/assign @smarterclayton @lavalamp @dchen1107 @thockin @brendandburns @jbeda 
/cc @grodrigues3 

Please ensure that every sub directory of / contains an owners file.

The following sub directories do not define them:
```
k8s.io/kubernetes$ for i in $(ls | grep -v -E '^_'); do if [[ ! -d $i ]]; then continue; elif [[ -e $i/OWNERS ]]; then continue; fi; echo MISSING OWNERS: $i; done
MISSING OWNERS: api
MISSING OWNERS: Godeps
MISSING OWNERS: hooks
MISSING OWNERS: staging
MISSING OWNERS: third_party
MISSING OWNERS: translations
MISSING OWNERS: vendor
```

The result of this is that any PR which changes anything in api, Godeps, staging, vendor, etc will always get assigned to one of you for both the review and approval.

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
This adds OWNERS files for the remaining subdirs of kubernetes/

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #44420

**Special notes for your reviewer**:
- translations: the kubectl specific translations are being handled by sig-cli, but the rest of the files like README and test have been handled and approved mostly by @brendandburns.
- third_party: this is a tricky mix of stuff like forked code.. mostly handled by the likes of @thockin, @lavalamp, or @smarterclayton for approvals.
- Godep/vendor: Right now @thockin mostly approves these as there are some tricky pieces around ensuring the licenses are compliant. For these, I've created a vendor-(reviewers|approvers) aliases, and associated those with both the Godeps and vendor folders. Right now @thockin is the only person in them, but that can easily expand in the future if there is a documented process on what to look for when approving dependency bumps.

**Release note**:
```release-note
NONE
```

/assign thockin 
/cc fejta

### CODE REVIEW

File: OWNERS_ALIASES [hunk 1]
```diff
@@ -223,3 +223,7 @@ aliases:
     - piosz
     - jsafrane
     - jbeda
+  vendor-reviewers:
+    - thockin
```
Comment (thockin) [inline OWNERS_ALIASES]: uhh, I guess I should be looking for volunteers to help with this.  @cblecker ?

Comment (dims) [reply OWNERS_ALIASES]: @thockin i can help if you are still looking for volunteers

**Pull request merged**